src/http/client.ts

```diff
@@ -0,0 +1,60 @@
+import fetch, { RequestInfo } from "node-fetch";
+
+type RequestParams = Array<string> | undefined;
+type JsonRpcId = number | string;
+
+export interface JsonRpcRequest {
+    jsonrpc: string;
+    method: string;
+    params: RequestParams;
+    id?: JsonRpcId | null;
+}
+
+export interface JsonRpcError {
+    code: number;
+    message: string;
+    data?: any;
+}
+
+export interface JsonRpcResponse {
+    jsonrpc: string;
+    result?: string;
+    error?: JsonRpcError;
+    id?: JsonRpcId | null;
+}
+
+export class JsonRpcClient {
+    url: string;
+
+    constructor(url: string) {
+        this.url = url;
+    }
+
+    async post(method: string, params?: RequestParams): Promise<JsonRpcResponse> {
+        const id = Math.random().toString(16).substring(7);
```

Comment:
I would make the id an ever incrementing number so that it's half useful for debugging.